### PR TITLE
Issue #191 add redirects for dataitem datafeed

### PR DESCRIPTION
**Pull request by MufengNiu**

### CODE REVIEW

**Review by yangli0516 — CHANGES_REQUESTED** [review]
Please handle the following route patterns as well:

- `places/{uid}?format={json|csv|kml}`: redirect to `places/{uid}/{format}`
- `places/{uid}?format={json|csv|kml}&download=on`: redirect to `places/{uid}/{format}?download=on`

**Comment by MufengNiu** [comment]
> Please handle the following route patterns as well:
> 
> * `places/{uid}?format={json|csv|kml}`: redirect to `places/{uid}/{format}`
> * `places/{uid}?format={json|csv|kml}&download=on`: redirect to `places/{uid}/{format}?download=on`
> 
> Also change the links from the "Download" and "WS feed" to the latest URL pattern.

Updated

File: resources/views/ws/ghap/places/show.blade.php [hunk 1]
```diff
@@ -22,17 +22,22 @@
 
                 <a href="{{route('index')}}#advancedsearch" class="btn btn-secondary tlcmgreen" role="button" id="advancedsearch" title="Search within an area, exact or fuzzy matching and filter by attributes.">Advanced Search</a>
 
+                <!-- Adjust for route places/{id}/{format?} -->
+                @php
+                    $url = url()->full();
+                    $separator = (parse_url($url, PHP_URL_QUERY) == NULL) ? '?' : '&';
+                @endphp
```
Comment (yangli0516) [inline resources/views/ws/ghap/places/show.blade.php:29]: Please try to avoid using blocks of logics inside views.

File: resources/views/ws/ghap/places/show.blade.php [hunk 2]
```diff
@@ -22,17 +22,22 @@
...
                     </button>
                     <div class="dropdown-menu" aria-labelledby="downloadDropdown">
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=kml&download=on">KML</a>
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=csv&download=on">CSV</a>
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=json&download=on">GeoJSON</a>
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=rocrate&download=on">RO-Crate</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=kml&download=on">KML</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=csv&download=on">CSV</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=json&download=on">GeoJSON</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=rocrate&download=on">RO-Crate</a>
```
Comment (yangli0516) [inline resources/views/ws/ghap/places/show.blade.php:40]: Ok, now I see these links should work with multiple places result page as well. Please revert the change back as the redirection should handle that.

File: resources/views/ws/ghap/places/show.blade.php [hunk 3]
```diff
@@ -42,9 +47,9 @@
                         WS Feed
                     </button>
                     <div class="dropdown-menu" aria-labelledby="wsfeedDropdown">
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=kml">KML</a>
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=csv">CSV</a>
-                        <a class="dropdown-item grab-hover" href="{{url()->full()}}&format=json">GeoJSON</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=kml">KML</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=csv">CSV</a>
+                        <a class="dropdown-item grab-hover" href="{{ $url . $separator }}format=json">GeoJSON</a>
```
Comment (yangli0516) [inline resources/views/ws/ghap/places/show.blade.php:52]: Ok, now I see these links should work with multiple places result page as well. Please revert the change back as the redirection should handle that.

**Review by yangli0516 — CHANGES_REQUESTED** [review]

File: resources/views/ws/ghap/places/show.blade.php (hunk 3 above)
Comment (MufengNiu) [reply resources/views/ws/ghap/places/show.blade.php:52]: if we click the 'WS FEED' button at URL' places/{uid}' . the URL will be wrong: 'places/{uid}&format=json' as there are no parameters

**Review by yangli0516 — APPROVED** [review]
